install.sh

```diff
@@ -462,55 +481,53 @@ pushd .
   # #################################################
   # configure & build
   # #################################################
-  cmake_common_options=""
-  cmake_client_options=""
 
   if [[ "${build_static}" == true ]]; then
     if [[ "${build_cuda}" == true ]]; then
       printf "Static library not supported for CUDA backend.\n"
       exit 1
     fi
-    cmake_common_options="${cmake_common_options} -DBUILD_SHARED_LIBS=OFF"
+    cmake_common_options+=("-DBUILD_SHARED_LIBS=OFF")
     compiler="${rocm_path}/bin/hipcc" #force hipcc for static libs, g++ doesn't work
     printf "Forcing compiler to hipcc for static library.\n"
   fi
 
   # build type
   if [[ "${build_release}" == true ]]; then
     mkdir -p ${build_dir}/release/clients && cd ${build_dir}/release
-    cmake_common_options="${cmake_common_options} -DCMAKE_BUILD_TYPE=Release"
+    cmake_common_options+=("-DCMAKE_BUILD_TYPE=Release")
   elif [[ "${build_release_debug}" == true ]]; then
     mkdir -p ${build_dir}/release-debug/clients && cd ${build_dir}/release-debug
-    cmake_common_options="${cmake_common_options} -DCMAKE_BUILD_TYPE=RelWithDebInfo"
+    cmake_common_options+=("-DCMAKE_BUILD_TYPE=RelWithDebInfo")
   else
     mkdir -p ${build_dir}/debug/clients && cd ${build_dir}/debug
-    cmake_common_options="${cmake_common_options} -DCMAKE_BUILD_TYPE=Debug"
+    cmake_common_options+=("-DCMAKE_BUILD_TYPE=Debug")
   fi
 
   # cuda
   if [[ "${build_cuda}" == true ]]; then
-    cmake_common_options="${cmake_common_options} -DUSE_CUDA=ON"
+    cmake_common_options+=("-DUSE_CUDA=ON")
   else
-    cmake_common_options="${cmake_common_options} -DUSE_CUDA=OFF"
+    cmake_common_options+=("-DUSE_CUDA=OFF")
   fi
 
   # clients
   if [[ "${build_clients}" == true ]]; then
-    cmake_client_options="${cmake_client_options} -DBUILD_CLIENTS_SAMPLES=ON -DBUILD_CLIENTS_TESTS=ON -DBUILD_CLIENTS_BENCHMARKS=ON"
+    cmake_client_options+=("-DBUILD_CLIENTS_SAMPLES=ON -DBUILD_CLIENTS_TESTS=ON -DBUILD_CLIENTS_BENCHMARKS=ON")
```

Comment:
Almost. Take a look at how it was done in rocSOLVER. What you've done works, but the value of using an array is that you can correctly handle arguments that contain spaces. (Not that we do, but it becomes possible.)

When you add multiple arguments separated by spaces in a single array element, you then need to split on spaces later, which breaks the original reason for using the array.

Reply:
I tried to split up this string like in rocSOLVER, but it didn't seem to work properly for me. I can give it another try today.

Reply:
Looks like I missed the `[@]` where cmake_common_options and cmake_client_options are used. Separating the array elements with spaces seems to work now.